adaptive/learner/balancing_learner.py

```diff
@@ -111,38 +111,57 @@ def strategy(self, strategy):
             raise ValueError(
                 'Only strategy="loss_improvements", strategy="loss", or'
                 ' strategy="npoints" is implemented.')
+        self._points = {}  # reset the cache
 
     def _ask_and_tell_based_on_loss_improvements(self, n):
-        points = []
-        loss_improvements = []
+        chosen_points = []
+        chosen_loss_improvements = []
```

Comment:
Just `loss_improvements`?

Reply:
I call `loss_improvements` what is returned be `ask`, so always `loss_improvements = ask(n=1)[1]`.